### PR TITLE
WRP-19326: Updated node modules' major versions

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Agate repo and sampler have node modules that need major updates.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
For Agate root:
Updated `immer` dependency to v10.
There were no breaking changes affecting our code https://github.com/immerjs/immer/releases.

For Agate sampler:
Updated `babel-loader` dependency to v9.
There were no breaking changes affecting our code https://github.com/babel/babel-loader/releases.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
For Agate I have also updated minor versions of `enact/docs-utils` and `enact/ui-test-utils`.
For sampler, minimum required node version is v14.15.0 https://github.com/babel/babel-loader/releases/tag/v9.0.0. I have also updated minor versions of `enact/storybook-utils` and `dotenv`.

### Links
[//]: # (Related issues, references)
WRP-19326

### Comments
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### CODE REVIEW

**Review by adrian-cocoara-lgp — APPROVED** [review]

**Review by andreitirlalge — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 The following is a curated list of changes in the Enact agate module, newest changes on the top.
 
+## unreleased
+
+- Updated dependencies including `immer` to version `^10.0.2`.
+- Updated `babel-loader` version to `^9.1.2` for sampler.
+- Updated the minimum version of Node to `14.15.0`for sampler.
```
Comment (daniel-stoian-lgp) [inline CHANGELOG.md:9]: i don't think sampler updates need to be added to the changelog of the agate library

File: samples/sampler/package.json [hunk 2]
```diff
@@ -16,6 +16,9 @@
     "serve-qa": "start-storybook -p 8080 -c .qa-storybook --ci"
   },
   "license": "Apache-2.0",
+  "engines": {
+    "node": ">=14.15.0"
+  },
```
Comment (daniel-stoian-lgp) [inline samples/sampler/package.json:21]: Only for the cli related modules we need to add this this

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 The following is a curated list of changes in the Enact agate module, newest changes on the top.
 
+## unreleased
```
Comment (daniel-stoian-lgp) [inline CHANGELOG.md:5]:
```suggestion
## [unreleased]
```

**Review by daniel-stoian-lgp — APPROVED** [review]

**Review by hong6316 — APPROVED** [review]
👍

**Pull request merged**